train_hierarchialXC.py

```diff
@@ -61,41 +61,42 @@
 
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Train your model ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-all_iters = 0
-cur_no = 0
-ae = GenericAutoencoder(loader.get_features().shape[1], 0.25, args.input_ae_dim)
+input = loader.get_features()
 
-INP_REC_LOSS = []
-
-for epoch in range(args.epochs):
+if args.input_ae_dim > 0:
+    all_iters = 0
     cur_no = 0
-    for x, _ in iter(data_loader):
-        cur_no += x.size(0)
+    ae = GenericAutoencoder(input.shape[1], 0.25, args.input_ae_dim)
 
-        inp_ae_fp = ae.forward(x.float())
-        loss_inp_rec = F.mse_loss(inp_ae_fp, x.float())
+    INP_REC_LOSS = []
 
-        all_iters += 1
-        if all_iters % args.interval == 0:
-            print("{} / {} :: {} / {} - INP_REC_LOSS : {}\t"
-                  .format(epoch, args.epochs, cur_no, len_loader, round(loss_inp_rec.item(), 5)))
-        INP_REC_LOSS.append(loss_inp_rec.item())
+    for epoch in range(args.epochs):
+        cur_no = 0
+        for x, _ in iter(data_loader):
+            cur_no += x.size(0)
 
-x = loader.get_features().toarray()
-x = ae.forward(torch.from_numpy(x).float())
-x = csr_matrix(x.detach().numpy())
+            inp_ae_fp = ae.forward(x.float())
+            loss_inp_rec = F.mse_loss(inp_ae_fp, x.float())
 
-start_time = time.time()
+            all_iters += 1
+            if all_iters % args.interval == 0:
+                print("{} / {} :: {} / {} - INP_REC_LOSS : {}\t"
+                      .format(epoch, args.epochs, cur_no, len_loader, round(loss_inp_rec.item(), 5)))
+            INP_REC_LOSS.append(loss_inp_rec.item())
 
-my_hierarchical_XC.train(x, loader.get_classes(), multiclass.OneVsRestClassifier,
-                         estimator=svm.SVC(), n_jobs=-1)
+    input = input.toarray()
+    input = ae.forward(torch.from_numpy(input).float())
```

Comment:
Refer to my old comment please. For large datasets, a full forward is not possible.

And also remove the `.float()` calls. This has been fixed in master.